packages/website/docs/using-react.md

```diff
@@ -0,0 +1,127 @@
+
+ All that you need to render is a `div` with the `containerRef` as the `ref`.
+
+```jsx title="Autocomplete.jsx"
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+### Mounting the autocomplete
+
+Now that you have access to the DOM through the `containerRef` object, you can create and mount the autocomplete instance.
+
+You can use the same [Autocomplete options](basic-options), but you must also pass the `renderer` and `render` parameters. You can rely on `props` to pass any autocomplete options you want to remain configurable.
+
+Don't forget to [clean up the effect](https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect) by returning a function that destroys the autocomplete instance.
+
+```jsx title="Autocomplete.jsx"
+import { autocomplete } from '@algolia/autocomplete-js';
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+
+    const search = autocomplete({
+      container: containerRef.current,
+      placeholder: 'Search',
+      renderer: { createElement, Fragment },
+      render({ children }, root) {
+        render(children, root);
+      },
+      ...props,
+    });
+
+    return () => {
+      search.destroy();
+    };
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+## Using the component
+
+Now that you've created an `<Autocomplete/>` component, you can use it in your React application.
+
+The example component sets the [`placeholder`](autocomplete-js#placeholder) and [`container`](autocomplete-js/#container) options, but lets everything else get configured through props.
+
+The example below sets [`openOnFocus`](autocomplete-js#openonfocus) and [sources](sources) (via [`plugins`](plugins)) through props. This example uses [recent searches](adding-recent-searches) as a [source](sources), but you could use anything else you want.
+
+The  Autocomplete library provides the [`createLocalStorageRecentSearchesPlugin`](createlocalstoragerecentsearchesplugin) function for creating a recent searches [plugin](plugins) out-of-the-box. To use it, you need to provide a `key` and `limit`.
```

Comment:
I think it would make more sense not to use a plugin here, but rather to use `getSources`. This would be the opportunity for us to explain that the last step for Autocomplete to work with React is to provide `createElement` to all highlighting utils (because it defaults to `preact.createElement`.

```jsx
import { highlightHit } from '@algolia/autocomplete-js';
import React, { createElement } from 'react';

function AutocompleteItem({ hit }) {
  return (
    <a href={hit.url} className="aa-ItemLink">
      <div className="aa-ItemContent">
        <div className="aa-ItemTitle">
          {highlightHit({
            hit,
            attribute: 'name',
            createElement,
          })}
        </div>
      </div>
    </a>
  );
}
```

(This applies for `highlightHit`, `snippetHit`, ` reverseHighlightHit` and `reverseSnippetHit`.)

And finally, add a note that all officially supported plugins work with React.